test/test_linalg.py

```diff
@@ -54,6 +58,430 @@ def test_det(self, device, dtype):
+                (-2, r'"svd_cuda" not implemented for \'Complex'),
+                ('nuc', r'"svd_cuda" not implemented for \'Complex'),
+            ]
+
+        # Test supported ords
+        for keepdim in [False, True]:
+            # vector norm
+            x = torch.randn(25, device=device, dtype=dtype)
+            xn = x.cpu().numpy()
+            for ord in supported_vector_ords:
+                res = torch.linalg.norm(x, ord, keepdim=keepdim).cpu()
+                expected = np.linalg.norm(xn, ord, keepdims=keepdim)
+                msg = gen_error_message(x.size(), ord, keepdim)
+                self.assertEqual(res.shape, expected.shape, msg=msg)
+                self.assertEqual(res, expected, msg=msg)
+
+            # matrix norm
+            x = torch.randn(25, 25, device=device, dtype=dtype)
+            xn = x.cpu().numpy()
+            for ord in supported_matrix_ords:
+                # TODO: Need to fix abort when nuclear norm is given cdouble input:
+                #       "double free or corruption (!prev) Aborted (core dumped)"
+                if ord == 'nuc' and dtype == torch.cdouble:
+                    continue
+                res = torch.linalg.norm(x, ord, keepdim=keepdim).cpu()
+                expected = np.linalg.norm(xn, ord, keepdims=keepdim)
+                msg = gen_error_message(x.size(), ord, keepdim)
+                self.assertEqual(res.shape, expected.shape, msg=msg)
+                self.assertEqual(res, expected, msg=msg)
+
+        # Test unsupported ords
+        # vector norm
+        x = torch.randn(25, device=device, dtype=dtype)
+        for ord, error_msg in unsupported_vector_ords:
+            with self.assertRaisesRegex(RuntimeError, error_msg):
+                torch.linalg.norm(x, ord)
+
+        # matrix norm
+        x = torch.randn(25, 25, device=device, dtype=dtype)
+        for ord, error_msg in unsupported_matrix_ords:
+            with self.assertRaisesRegex(RuntimeError, error_msg):
+                torch.linalg.norm(x, ord)
+
+    # TODO: When integer dims are supported in norm, remove this test
+    def test_norm_dim_int_error(self, device):
+        input = torch.randn(10, device=device)
+        with self.assertRaisesRegex(TypeError, r'linalg_norm\(\) received an invalid combination of arguments'):
+            torch.linalg.norm(input, dim=0)
+
+    @skipCUDAIfNoMagma
+    @skipCPUIfNoLapack
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_norm_extreme_values(self, device):
+        vector_ords = [0, 1, 2, 3, inf, -1, -2, -3, -inf]
+        matrix_ords = ['fro', 'nuc', 1, 2, inf, -1, -2, -inf]
+        vectors = []
+        matrices = []
+        for pair in itertools.product([inf, -inf, 0.0, nan], repeat=2):
+            vectors.append(list(pair))
+            matrices.append([[pair[0], pair[1]]])
+            matrices.append([[pair[0]], [pair[1]]])
+
+        # TODO: The following ord and vector combinations do not produce matching results in
```

Comment:
I was able to fix this problem by just implementing my own `ord = +/- inf` vector norm that calculates `input.abs().min/max()`. This might not be as fast as the `at::norm` function (which combines everything into one kernel dispatch), but at least it gives the correct result. This solution also helped me fix a couple other issues with how the `ord = +/- inf`vector norm was handling degenerate shapes.